library/src/test/java/bio/terra/landingzone/stairway/flight/create/resource/step/CreateBatchAccountStepTest.java

```diff
@@ -89,6 +96,34 @@ void doStepSuccess() throws InterruptedException {
     verifyNoMoreInteractions(mockBatchAccountDefinitionStagesWithCreate);
   }
 
+  @Test
+  void doStepWhenQuotaIssueThrown() throws InterruptedException {
+    TargetManagedResourceGroup mrg = ResourceStepFixture.createDefaultMrg();
+    when(mockResourceNameProvider.getName(createBatchAccountStep.getResourceType()))
+        .thenReturn(BATCH_ACCOUNT_NAME);
+
+    setupFlightContext(
+        mockFlightContext,
+        Map.of(
+            LandingZoneFlightMapKeys.BILLING_PROFILE,
+            new ProfileModel().id(UUID.randomUUID()),
+            LandingZoneFlightMapKeys.LANDING_ZONE_ID,
+            LANDING_ZONE_ID,
+            LandingZoneFlightMapKeys.LANDING_ZONE_CREATE_PARAMS,
+            ResourceStepFixture.createLandingZoneRequestForCromwellLandingZone()),
+        Map.of(GetManagedResourceGroupInfo.TARGET_MRG_KEY, mrg));
+    setupMocks(BATCH_ACCOUNT_NAME, mrg.region(), mrg.name());
+    var batchAccountQuotaException = mockBatchQuotaException();
+    doThrow(batchAccountQuotaException).when(mockBatchAccountDefinitionStagesWithCreate).create();
+
+    var stepResult = createBatchAccountStep.doStep(mockFlightContext);
+
+    assertThat(stepResult.getStepStatus(), equalTo(StepStatus.STEP_RESULT_FAILURE_FATAL));
+    assertTrue(stepResult.getException().isPresent());
+    assertTrue(stepResult.getException().get() instanceof LandingZoneCreateException);
+    assertNotNull(stepResult.getException().get().getCause());
```

Comment:
So is line 124 what ensures we have the original exception message? Would it be possible to make it clearer by making sure that "The regional Batch account quota for the specified subscription has been reached" appears in the exception information?

Reply:
L124 is for the original exception.